core/src/main/java/org/apache/carbondata/core/keygenerator/mdkey/NumberCompressor.java

```diff
@@ -137,12 +161,31 @@ public NumberCompressor(int cardinality) {
     return words;
   }
 
+  //  public int[] unCompress(byte[] key, int offset, int length) {
+  //    int ls = length;
+  //    int arrayLength = (ls * BYTE_LENGTH) / bitsLength;
+  //    long[] words = new long[getWordsSizeFromBytesSize(ls)];
+  //    unCompressVal(key, ls, words, offset);
+  //    return getArray(words, arrayLength);
+  //  }
+
   public int[] unCompress(byte[] key, int offset, int length) {
-    int ls = length;
-    int arrayLength = (ls * BYTE_LENGTH) / bitsLength;
-    long[] words = new long[getWordsSizeFromBytesSize(ls)];
-    unCompressVal(key, ls, words, offset);
-    return getArray(words, arrayLength);
+    int len = length / bytesPerKey;
+    if (length % bytesPerKey != 0) {
+      throw new IllegalArgumentException(
+          "Some thing wrong while decompress" + length + " " + bytesPerKey);
```

Comment:
Both length is read from footer, right? If comes here, mean the file is corrupted.